backend/src/main/java/com/redhat/cloud/notifications/db/repositories/TemplateRepository.java

```diff
@@ -295,4 +298,46 @@ private Template findTemplate(UUID id, String notFoundMessage) {
         }
         return template;
     }
+
+
+    public Optional<GenericTemplate> findGenericTemplate(String appName,
+                                                         String accountId,
+                                                         GenericTemplate.TemplateKind templateKind,
+                                                         String integrationType) {
+        String hql = "FROM GenericTemplate gt JOIN FETCH gt.theTemplate " +
+                "WHERE gt.templateKind <= :templateKind " +
+                "AND gt.integrationType = :iType " +
+                (accountId != null ? "AND gt.accountId = :accountId " : " ") +
+                (appName != null ? "AND gt.application.name = :appName " : " ") +
+                "ORDER BY gt.templateKind DESC ";
+        Query query = entityManager.createQuery(hql, GenericTemplate.class)
+                .setParameter("templateKind", templateKind)
+                .setParameter("iType", integrationType)
+                .setMaxResults(1);
+
+        if (appName != null) {
+            query.setParameter("appName", appName);
+        }
+        if (accountId != null) {
+            query.setParameter("accountId", accountId);
+        }
+
+        List<GenericTemplate> templates = query.getResultList();
+
+        if (templates.isEmpty()) {
+            if (templateKind.equals(GenericTemplate.TemplateKind.USER) && accountId != null) {
```

Comment:
For the other ones it would already fall back by the way it is done. for USER (now ORG) we need to remove the accountId before feeding back into the query for the fallbacks